airflow/models/skipmixin.py

```diff
@@ -121,8 +121,11 @@ def skip(
             raise ValueError("dag_run is required")
 
         task_ids_list = [d.task_id for d in task_list]
-        SkipMixin._set_state_to_skipped(dag_run, task_ids_list, session)
-        session.commit()
+
+        # The following could be applied only for non-mapped tasks
+        if map_index == -1:
+            SkipMixin._set_state_to_skipped(dag_run, task_ids_list, session)
+            session.commit()
```

Comment:
This block could theoretically be removed, and then setting the skipped states will be done exclusively by `NotPreviouslySkippedDep` for both mapped and none-mapped.
Not sure regarding effects on performance, so I left it as-is for now (also, it affects some other tests).